storages/backends/azure_storage.py

```diff
@@ -93,9 +88,9 @@ def _save(self, name, content):
         else:
             content_data = content.read()
 
-        self.connection.put_blob(self.azure_container, name,
-                                 content_data, "BlockBlob",
-                                 x_ms_blob_content_type=content_type)
+        self.connection.create_blob_from_bytes(self.azure_container, name,
+                                 content_data,
+                                 content_settings=ContentSettings(content_type=content_type))
```

Comment:
This still reads everything into memory in the `if/else` above. I think if we just pass the file to `create_blob_from_file` it might _just work_.